resource-providers/src/main/java/io/opentelemetry/resource/spring/SpringBootServiceNameGuesser.java

```diff
@@ -0,0 +1,255 @@
+            this::findByCurrentDirectoryApplicationProperties,
+            this::findByClasspathApplicationYaml,
+            this::findByCurrentDirectoryApplicationYaml,
+            this::findByCommandlineArgument);
+    return finders
+        .map(Supplier::get)
+        .filter(Objects::nonNull)
+        .findFirst()
+        .map(
+            serviceName -> {
+              logger.debug("Guessed Spring Boot service name: " + serviceName);
+              return Resource.builder().put(ResourceAttributes.SERVICE_NAME, serviceName).build();
+            })
+        .orElseGet(Resource::empty);
+  }
+
+  @Nullable
+  private String findByEnvironmentVariable() {
+    String result = system.getenv("SPRING_APPLICATION_NAME");
+    logger.debug("Checking for SPRING_APPLICATION_NAME in env: " + result);
+    return result;
+  }
+
+  @Nullable
+  private String findBySystemProperties() {
+    String result = system.getProperty("spring.application.name");
+    logger.debug("Checking for spring.application.name system property: " + result);
+    return result;
+  }
+
+  @Nullable
+  private String findByClasspathApplicationProperties() {
+    String result = readNameFromAppProperties();
+    logger.debug("Checking for spring.application.name in application.properties file: " + result);
+    return result;
+  }
+
+  @Nullable
+  private String findByCurrentDirectoryApplicationProperties() {
+    String result = null;
+    try (InputStream in = system.openFile("application.properties")) {
+      result = getAppNamePropertyFromStream(in);
+    } catch (Exception e) {
+      // expected to fail sometimes
+    }
+    logger.debug("Checking application.properties in current dir: " + result);
+    return result;
+  }
+
+  @Nullable
+  private String findByClasspathApplicationYaml() {
+    String result =
+        loadFromClasspath("application.yml", SpringBootServiceNameGuesser::parseNameFromYaml);
+    logger.debug("Checking application.yml in classpath: " + result);
+    return result;
+  }
+
+  @Nullable
+  private String findByCurrentDirectoryApplicationYaml() {
+    String result = null;
+    try (InputStream in = system.openFile("application.yml")) {
+      result = parseNameFromYaml(in);
+    } catch (Exception e) {
+      // expected to fail sometimes
+    }
+    logger.debug("Checking application.yml in current dir: " + result);
+    return result;
+  }
+
+  @Nullable
+  @SuppressWarnings("unchecked")
+  private static String parseNameFromYaml(InputStream in) {
+    Yaml yaml = new Yaml();
+    try {
+      Map<String, Object> data = yaml.load(in);
+      Map<String, Map<String, Object>> spring =
+          (Map<String, Map<String, Object>>) data.get("spring");
+      if (spring != null) {
+        Map<String, Object> app = spring.get("application");
+        if (app != null) {
+          Object name = app.get("name");
+          if (name != null) {
+            return (String) name;
+          }
```

Comment:
No need for a null check here:
```suggestion
          return (String) app.get("name");
```